src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -90,17 +91,34 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
         return BlobInfo.newBuilder(getBlobId(fileId, info)).build()
     }
 
+    private fun cleanUpLog() {
+        MDC.remove("funName")
+        MDC.remove("projectId")
+        MDC.remove("fileId")
+        MDC.remove("userId")
+    }
+
+    private fun putMVC(funName: String, projectId: String, fileId: String = "", userId: String = "") {
```

Comment:
I suggest re-writing `putMVC` so that it accepted lambda in the last argument and called it after filling out MDC, and then called cleanup. And renaming to `logging`:
```
logging(projectId = ..., fileId = ...) {
   // do stuff
}
```

This way you will never forget to call `cleanupLog`

Reply:
done